### PR TITLE
refactor(site): simplify workspace topbar

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
From @mafredri feedback:
- It looks a bit busy in my eyes, I think it could be simplified further.
- Maybe we can change the beginning to [template icon] [username] / [workspace] (i) [Stops in ...]. My thought is that behind (i) (info) we can show template name, current version, etc.
-  the current version of the workspace/template is quite uninteresting and only needed in special circumstances.

Changes:
- Moved template name and version into a tooltip when hovering the template icon
- Moved the user name to be the first info in the "workspace path"
- Removed the outdated pill as well since we already have on the page an update button and alert with this information.

Before:
Screenshot 2024-01-02 at 14 24 34

After:
Screenshot 2024-01-02 at 14 50 34

### CODE REVIEW

**Review by matifali — APPROVED** [review]

File: site/src/pages/WorkspacePage/WorkspaceTopbar/WorkspaceTopbar.tsx [hunk 1]
```diff
@@ -120,58 +120,77 @@ export const WorkspaceTopbar = (props: WorkspaceProps) => {
...
           <TopbarDivider />
-          <Link
-            component={RouterLink}
-            to={`/templates/${workspace.template_name}`}
-            css={{ color: "inherit" }}
+          <Popover
+            mode="hover"
+            // title={`${
+            //   workspace.template_display_name ?? workspace.template_name
+            // } on version ${workspace.latest_build.template_version_name}`}
```
Comment (mafredri) [inline site/src/pages/WorkspacePage/WorkspaceTopbar/WorkspaceTopbar.tsx:134]: Is the commented code intentional? Will it be used later?

**Review by mafredri — APPROVED** [review]
Love it, great work! ❤️

File: site/src/pages/WorkspacePage/WorkspaceTopbar/WorkspaceTopbar.tsx (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/pages/WorkspacePage/WorkspaceTopbar/WorkspaceTopbar.tsx:134]: No, ty!

**Comment by BrunoQuaresma** [comment]
Related to https://github.com/coder/coder/issues/11491

**Pull request merged**